server/src-lib/Data/Sequence/NonEmpty.hs

```diff
@@ -31,3 +34,9 @@ v <| (NESeq (h, l)) = NESeq (v, h Seq.<| l)
 
 toSeq :: NESeq a -> Seq.Seq a
 toSeq (NESeq (v, l)) = v Seq.<| l
+
+toList :: NESeq a -> [a]
+toList = F.toList . toSeq
+
+length :: NESeq a -> Int
+length = Seq.length . toSeq
```

Comment:
Might it be better to implement `Foldable` for `NESeq`? The implementation would be pretty easy, since you can get away with just implementing `foldr`, which you could do as just `foldr f v = F.foldr f v . toSeq`. Then you’d get `toList` and `length` for free, since those work on any `Foldable`. 😄